lib/carma/models/submission/metadata.rb

```diff
@@ -4,12 +4,42 @@ module CARMA
   module Models
     class Submission
       class Metadata < CARMA::Models::Base
-        request_payload_key :claim_id
+        request_payload_key :claim_id,
+                            :veteran,
+                            :primary_caregiver,
+                            :secondary_caregiver_one,
+                            :secondary_caregiver_two
 
-        attr_reader :claim_id
+        attr_accessor       :claim_id
+
+        attr_reader         :veteran,
+                            :primary_caregiver,
+                            :secondary_caregiver_one,
+                            :secondary_caregiver_two
 
         def initialize(args = {})
           @claim_id = args[:claim_id]
+
+          self.veteran = args[:veteran] || {}
+          self.primary_caregiver = args[:primary_caregiver] || {}
+          self.secondary_caregiver_one = args[:secondary_caregiver_one]
+          self.secondary_caregiver_two = args[:secondary_caregiver_two]
+        end
+
+        def veteran=(veteran_data_hash)
+          @veteran = Metadata::Veteran.new(veteran_data_hash)
+        end
+
+        def primary_caregiver=(pc_metadata_hash)
+          @primary_caregiver = Metadata::Caregiver.new(pc_metadata_hash)
+        end
+
+        def secondary_caregiver_one=(sc_one_metadata_hash)
+          @secondary_caregiver_one = Metadata::Caregiver.new(sc_one_metadata_hash) unless sc_one_metadata_hash.nil?
```

Comment:
I think this could have the weird side effect that if it's set to a value, and then later someone tries to set it as nil it won't change the value.  But I don't imagine that's a use-case for this

Reply:
Good catch. I've updated it.